sympy/functions/elementary/piecewise.py

```diff
@@ -546,6 +563,17 @@ def piecewise_fold(expr):
         for e, c in expr.args:
             if not isinstance(e, Piecewise):
                 e = piecewise_fold(e)
+            if c.has(Piecewise):
+                c = piecewise_fold(c)
+                if isinstance(c, Piecewise):
+                    # Piecewise((c1, k1), (c2, k2)) ->
```

Comment:
ping @jksuom , can you give these few lines a once over, please? I am rewriting a Piecewise condition as an Or or Ands, e.g. `Piecewise((x < 1, y < 0), (x > 2, y < 1))` -> `(x < 1) & (y < 0) | (x > 2) & (y < 1) & ~(y < 0)`.

Reply:
Perhaps this should become a method of Piecewise so one could write a Piecewise of conditions and then convert it into a Boolean. Perhaps `Piecewise.as_boolean()` or `to_nnf` as for `ITE`?

Reply:
But I think this is the `dnf` form, so the method would be `to_dnf`.